src/packages/core/property-editor/uis/tree-picker/config/filter/property-editor-ui-tree-picker-filter.element.ts

```diff
@@ -0,0 +1,95 @@
+import { UMB_DATA_TYPE_WORKSPACE_CONTEXT } from '../../../../../data-type/workspace/data-type-workspace.context.js';
+import { UmbDocumentTypeInputElement } from '../../../../../../documents/document-types/components/document-type-input/document-type-input.element.js';
+import { UmbMediaTypeInputElement } from '../../../../../../media/media-types/components/media-type-input/media-type-input.element.js';
+import { StartNode } from '@umbraco-cms/backoffice/components';
+import type { UmbPropertyEditorUiElement } from '@umbraco-cms/backoffice/extension-registry';
+import { customElement, html, property } from '@umbraco-cms/backoffice/external/lit';
+import { UmbTextStyles } from '@umbraco-cms/backoffice/style';
+import { UmbLitElement } from '@umbraco-cms/internal/lit-element';
+
+@customElement('umb-property-editor-ui-tree-picker-filter')
+export class UmbPropertyEditorUITreePickerFilterElement extends UmbLitElement implements UmbPropertyEditorUiElement {
+	constructor() {
+		super();
+
+		this.consumeContext(UMB_DATA_TYPE_WORKSPACE_CONTEXT, (instance) => {
```

Comment:
Actually, just to be super duper correct, then there is no need to utilize the Workspace in this case.

Not that the perspective I will mention now does not have any effect on this case, but I wanted to let you know, for your general knowledge of this:

So we have a property-dataset-context, which is more local than the workspace. Its generic and will always be present when we utilise property editor UIs. (Where a Workspace is not necessary, especially the specific version (Data-type workspace))

So again, this does not make any difference since this Property Editor UIs is very specific to this specific Data-Type.

But if you used the Property Dataset Context instead, then this Property Editor UI could be used in any other scenario, as long as there is another property with the alias StartNode (that matches that data-structure)

let me know if you like these insights or if its just noise :-)

Reply:
@nielslyngsoe Understood. 👍  This helps to answer my previous question (above), https://github.com/umbraco/Umbraco.CMS.Backoffice/pull/1083#discussion_r1439350482.